.github/workflows/update-plugins-repo-refs.yaml

```
@@ -63,243 +68,293 @@ jobs:
           INPUT_OVERLAY_REPO: ${{ inputs.overlay-repo }}
           INPUT_RELEASE_BRANCH_PATTERN: ${{ inputs.release-branch-pattern }}
           INPUT_WORKSPACE_PATH: ${{ inputs.workspace-path }}
+          INPUT_SINGLE_BRANCH: ${{ inputs.single-branch }}
+          INPUT_VERBOSE: ${{ inputs.verbose }}
+          INPUT_DEBUG: ${{ inputs.debug }}
           GH_TOKEN: ${{ github.token }}
 
         run: |
           # Plugin Retrieval
 
+          function message() {
+            echo "$1" >&2
+          }
+
           function verbose() {
-            if [[ "${{ inputs.verbose }}" == "true" ]]
+            if [[ "${INPUT_VERBOSE}" == "true" ]]
             then
-              echo "$1"
+              message "$1"
             fi
           }
 
-          if [[ "${{ inputs.debug }}" == "true" ]]
+          if [[ "${INPUT_DEBUG}" == "true" ]]
           then
             set -x
           fi
 
           set -o pipefail
           npm install semver -g
-          regexps="${INPUT_REGEXPS}"
-          overlayRepoBranchesString=$(gh api repos/${INPUT_OVERLAY_REPO}/branches | jq -r --arg regex "${INPUT_RELEASE_BRANCH_PATTERN}" '.[].name | select(test($regex))')
+          overlayRepoBranchesString=$(gh api repos/${INPUT_OVERLAY_REPO}/branches | jq -r --arg regex "${INPUT_RELEASE_BRANCH_PATTERN}" --arg singleBranchPattern "^${INPUT_SINGLE_BRANCH}$" '.[].name | select(test($regex) or test($singleBranchPattern))')
```

just curious why the need to rename variables in your json query? why not just... 
```suggestion
          overlayRepoBranchesString=$(gh api repos/${INPUT_OVERLAY_REPO}/branches | jq -r --arg INPUT_RELEASE_BRANCH_PATTERN "${INPUT_RELEASE_BRANCH_PATTERN}" --arg INPUT_SINGLE_BRANCH"^${INPUT_SINGLE_BRANCH}$" '.[].name | select(test($INPUT_RELEASE_BRANCH_PATTERN) or test($INPUT_SINGLE_BRANCH))')
```

to avoid obfuscating variables that you only use once?